ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -371,7 +373,10 @@ void read(DataStreamRequestByteBuf request, ChannelHandlerContext ctx,
       // for peers to start transaction
       final StreamInfo info = streams.get(key);
       composeAsync(info.getPrevious(), executor, v -> startTransaction(info, request, ctx))
-          .thenAccept(v -> buf.release());
+          .thenAccept(v -> buf.release()).exceptionally(exception -> {
+        replyDataStreamException(server, exception, info.getRequest(), request, ctx);
+        return null;
+      });
```

Comment:
Use whenComplete(..) and also a try-catch as below:
```suggestion
          .whenComplete((v, exception) -> {
        try {
          if (exception != null) {
            replyDataStreamException(server, exception, info.getRequest(), request, ctx);
          }
        } finally {
          buf.release();
        }
      });
```